import_export/admin.py

```diff
@@ -265,7 +270,12 @@ def import_action(self, request, *args, **kwargs):
 
             # prepare kwargs for import data, if needed
             res_kwargs = self.get_import_resource_kwargs(request, form=form, *args, **kwargs)
-            resource = self.get_import_resource_class()(**res_kwargs)
+            if request.POST.get('resource', None):
+                resource_class = self.get_import_resource_class()[int(request.POST['resource'])]
```

Comment:
This line is a duplicate of line 123, therefore I think it would be better if there was a separate method to safely read the resource_class based on the value in POST.  Since, this could be an untrusted value sent via a crafted POST request, I think there should be error checking to check that it will pick a valid index within the `resource_class` list.  At present, if I were to send a value outside of the length of the error it would crash.

Reply:
Good point about the code duplication. I fixed that.

It would have not crash, because the form didn't let it through (I tested). Anyway, I changed it, so it takes the cleaned value from the form to make the code cleaner.

Reply:
Fantastic.  I think it would be ideal to have a unit test for `choose_import_resource_class()`

Reply:
I added some unit test for that methods.